report/thermodesorption-spectra/ogorodnikova-tds.ipynb

```diff
@@ -0,0 +1,299 @@
+      "text/plain": [
+       "<Figure size 640x480 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "t = derived_quantities.t\n",
+    "flux_left = derived_quantities.filter(fields=\"solute\", surfaces=1).data\n",
+    "flux_right = derived_quantities.filter(fields=\"solute\", surfaces=2).data\n",
+    "flux_total = -np.array(flux_left) - np.array(flux_right)\n",
+    "\n",
+    "trap_1 = derived_quantities.filter(fields=\"1\").data\n",
+    "trap_2 = derived_quantities.filter(fields=\"2\").data\n",
+    "trap_3 = derived_quantities.filter(fields=\"3\").data\n",
+    "\n",
+    "contribution_trap_1 = -np.diff(trap_1)/np.diff(t)\n",
+    "contribution_trap_2 = -np.diff(trap_2)/np.diff(t)\n",
+    "contribution_trap_3 = -np.diff(trap_3)/np.diff(t)\n",
+    "\n",
+    "temp = [implantation_temp + 8*(t - start_tds) for t in t]\n",
```

Comment:
If you convert `t` to a numpy `array` you can simplify this to:

```python
t = np.array(t)
temp = implantation_temp + 8*(t - start_tds)
```

In general you want to do this to avoid looping through very large lists. Imagine if `t` has millions of elements, looping like you do here would be extremely slow.